tests/samples.ts

```diff
@@ -215,3 +139,118 @@ export const albMultiValHeadersRequest = (): ApplicationLoadBalancerRequestEvent
+      resourcePath: '/',
+   };
+};
+
+export const makeAPIGatewayRequestEventHeaders =
+   (params?: MakeAPIGatewayRequestEventHeadersParams): APIGatewayRequestEvent['headers'] => {
+      return withDefault<APIGatewayRequestEvent['headers']>(
+         params,
+         {
+            Accept: '*/*',
+            'CloudFront-Forwarded-Proto': 'https',
+            'CloudFront-Is-Desktop-Viewer': 'true',
+            'CloudFront-Is-Mobile-Viewer': 'false',
+            'CloudFront-Is-SmartTV-Viewer': 'false',
+            'CloudFront-Is-Tablet-Viewer': 'false',
+            'CloudFront-Viewer-Country': 'US',
+            Host: 'b5gee6dacf.execute-api.us-east-1.amazonaws.com',
+            'User-Agent': 'curl/7.54.0',
+            Via: '2.0 4ee511e558a0400aa4b9c1d34d92af5a.cloudfront.net (CloudFront)',
+            'X-Amz-Cf-Id': 'xn-ohXlUAed-32bae2cfb7164fd690ffffb87d36b032==',
+            'X-Amzn-Trace-Id': 'Root=1-4b5398e2-a7fbe4f92f2e911013cba76b',
+            'X-Forwarded-For': '8.8.8.8, 2.3.4.5',
+            'X-Forwarded-Port': '443',
+            'X-Forwarded-Proto': 'https',
+            Referer: 'https://en.wikipedia.org/wiki/HTTP_referer',
+            Cookie: 'uid=abc; ga=1234; foo=bar; baz=foo%5Ba%5D; obj=j%3A%7B%22abc%22%3A123%7D; onechar=j; bad=j%3A%7Ba%7D',
+         }
+      );
+   };
+
+export const makeAPIGatewayRequestEventMultiValueHeader =
+   (params?: MakeAPIGatewayRequestEventMultiValueHeadersParams): APIGatewayRequestEvent['multiValueHeaders'] => {
+      return withDefault<APIGatewayRequestEvent['multiValueHeaders']>(
+         params,
+         {
+            Foo: [ 'bar', 'baz' ],
+            ...Object.fromEntries(
+               Object.entries(makeAPIGatewayRequestEventHeaders())
+                  .map(([ key, value ]): Array<string | Array<string | undefined>> => {
+                     return [ key, [ value ] ];
+                  })
+            ),
+         }
+      );
+   };
+
+export const makeAPIGatewayRequestEvent = (params?: MakeAPIGatewayRequestEventParams): APIGatewayRequestEvent => {
+   const httpMethod = withDefault(params?.httpMethod, 'GET'),
+         path = withDefault(params?.path, '/echo/asdf/a'),
+         headers = withDefault(params?.headers, makeAPIGatewayRequestEventHeaders());
+
+   const multiValueHeaders = withDefault(
+      params?.multiValueHeaders,
+      makeAPIGatewayRequestEventMultiValueHeader()
+   );
+
+   return {
+      path,
+      httpMethod,
+      body: null,
+      isBase64Encoded: false,
+      resource: '/{proxy+}',
+      pathParameters: { proxy: path },
+      stageVariables: null,
+      requestContext: makeAPIGatewayRequestContext({
+         httpMethod,
+      }),
+      headers,
+      multiValueHeaders,
+      queryStringParameters: {
+         'foo[a]': 'bar b',
+         x: '2',
+         y: 'z',
+      },
+      multiValueQueryStringParameters: {
+         'foo[a]': [ 'bar b', 'baz c' ],
+         x: [ '1', '2' ],
+         y: [ 'z' ],
+      },
+   };
+};
```

Comment:
I've also thought of another approach for handling default values by using the object behavior of making keys that comes at last overwrite the preceding ones:

```js
// say params is:
const params = {
   path: '/another/path/a',
   httpMethod: 'POST',
};
```

```js
export const makeAPIGatewayRequestEvent = (params?: MakeAPIGatewayRequestEventParams): APIGatewayRequestEvent => {
   // still treating these values because in the default case the must reflected in other
   // parts of the object
   const httpMethod = withDefault(params?.httpMethod, 'GET'),
         path = withDefault(params?.path, '/echo/asdf/a');
   return {
      path: '/echo/asdf/a',
      httpMethod,
      body: null,
      isBase64Encoded: false,
      resource: '/{proxy+}',
      pathParameters: { proxy: path },
      stageVariables: null,
      requestContext: makeAPIGatewayRequestContext({
         httpMethod,
      }),
      headers: makeAPIGatewayRequestEventHeaders(),
      multiValueHeaders: makeAPIGatewayRequestEventMultiValueHeader(),
      queryStringParameters: {
         'foo[a]': 'bar b',
         x: '2',
         y: 'z',
      },
      multiValueQueryStringParameters: {
         'foo[a]': [ 'bar b', 'baz c' ],
         x: [ '1', '2' ],
         y: [ 'z' ],
      },
      
      ...params,
   };
};
```

It seems to have a similar behavior and it is easier for developers to add more of the dynamic fields in the future, since they only need to update the interface if no further treatment is needed like in the cases of `path` and `httpMethod`.